core/src/test/java/org/elasticsearch/action/bulk/BulkWithUpdatesIT.java

```diff
@@ -58,10 +58,44 @@
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.nullValue;
 
-public class BulkTests extends ESIntegTestCase {
+public class BulkWithUpdatesIT extends ESIntegTestCase {
+
     @Override
     protected Collection<Class<? extends Plugin>> nodePlugins() {
-        return Collections.singleton(GroovyPlugin.class);
+        return Collections.singleton(CustomScriptPlugin.class);
+    }
+
+    public static class CustomScriptPlugin extends MockScriptPlugin {
+
+        @Override
+        @SuppressWarnings("unchecked")
+        protected Map<String, Function<Map<String, Object>, Object>> pluginScripts() {
+            Map<String, Function<Map<String, Object>, Object>> scripts = new HashMap<>();
+
+            scripts.put("ctx._source.field += 1", vars -> {
+                Map<?, ?> ctx = (Map<?, ?>) vars.get("ctx");
+
+                Map<String, Object> source = (Map<String, Object>) ctx.get("_source");
+                Integer field = (Integer) source.get("field");
+                return source.replace("field", field + 1);
+            });
+
+            scripts.put("ctx._source.counter += 1", vars -> {
+                Map<?, ?> ctx = (Map<?, ?>) vars.get("ctx");
+
+                Map<String, Object> source = (Map<String, Object>) ctx.get("_source");
+                Integer counter = (Integer) source.get("counter");
+                return source.replace("counter", counter + 1);
+            });
+
+            scripts.put("ctx._source.field += a", vars -> {
```

Comment:
Another fun point: while groovy would throw an exception when the script runs painless would refuse to compile this.